config/routes.rb

```diff
@@ -0,0 +1,22 @@
+Rails.application.routes.draw do
+  # get 'users/index'
+  devise_for :users
+  unauthenticated :user do
+    root to: 'users#index'
+  end
+
+  authenticated :user do
+    root to: 'categories#index', as: :authenticated_root
+  end
+
+  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
+  resources :categories do
+    resources :transactions 
+  end
```

Comment:
- [ ] As requested by previous reviewer, you have to use only the necessary routes are used on this project. By default, Rails creates routes for all seven of the default actions (index, show, new, create, edit, update, and destroy) for every RESTful route in your application. You can use the `:only and : except` options to fine-tune this behavior. The :only option specifies that only certain routes should be generated. This can really save our memory space.for ex:-
```
 resources :users, only: %i[index new create  destroy ] do
    resources :posts, only: %i[index show new create update destroy]
  end
```